src/java/voldemort/store/rocksdb/RocksDbStorageConfiguration.java

```diff
@@ -34,13 +37,39 @@
 
     private Map<String, RocksDbStorageEngine> stores = new HashMap<String, RocksDbStorageEngine>();
 
+    // RocksDB Configuration Options
+    private final CompactionStyle compactionStyle;
+    private final CompressionType compressionType;
+    private final int maxBackgroundCompactions;
+    private final int maxBackgroundFlushes;
+    private final long maxBytesForLevelBase;
+    private final int maxWriteBufferNumber;
+    private final int statsDumpPeriodSec;
+    private final long targetFileSizeBase;
+    private final int targetFileSizeMultiplier;
+    private final long writeBufferSize;
+
     public RocksDbStorageConfiguration(VoldemortConfig config) {
         /**
          * - TODO 1. number of default locks need to debated. This default is
          * same as that of Krati's. 2. Later add the property to VoldemortConfig
          */
         this.voldemortconfig = config;
-        this.lockStripes = config.getAllProps().getInt("rocksdb.lock.stripes", 50);
+        Props props = config.getAllProps();
+
+        this.lockStripes = props.getInt("rocksdb.lock.stripes", 50);
+
+        // TODO: Validate the default mandatory options
+        compactionStyle = CompactionStyle.valueOf(props.getString("rocksdb.options.compactionStyle", CompactionStyle.UNIVERSAL.toString()));
```

Comment:
I saw the email notification for your earlier comment, but, also did not see it here.  I did find it associated with an obsolete commit:

https://github.com/jwlent55/voldemort/commit/6513d86

I think I may have caused the problem by amending my update to fix a spelling mistake just as you were commenting on the code.

In any event, happy to make the change.  I debated (with myself) where to place the option definitions: VoldemortConfig or RocksDbStorageConfiguration.  I ended up following the Krati pattern.  Figured I would guess wrong.